docs/reference/content/guides/querying.md

```diff
@@ -0,0 +1,268 @@
+
+{{% note %}}
+Regardless of the approach used, the field name given can be either the Java field name or the document field name as defined by the 
+`@Property` annotation on the field.  Morphia will normalize the name and validate the name such that a query with a bad field name will 
+result in an error.
+{{% /note %}}
+
+## Complex Queries
+
+Of course, queries are usually more complex than single field comparisons.  Morphia offers both `and()` and `or()` to build up more 
+complex queries.  An `and` query might look something like this:
+
+```java
+q.and(
+    q.criteria("width").equal(10),
+    q.criteria("height").equal(1)
+);
+```
+
+An `or` clause looks exactly the same except for using `or()` instead of `and()`, of course.  For these clauses we use the `criteria()` 
+method instead of `field()` but it is used in much the same fashion.  `and()` and `or()` take a [`varargs`](https://docs.oracle
+.com/javase/8/docs/technotes/guides/language/varargs.html) parameter of type `Criteria` so you can include as many filters as necessary. 
+ If all you need is an `and` clause, you don't need an explicit call to `and()`:
+ 
+```java
+datastore.createQuery(UserLocation.class)
+    .field("x").lessThan(5)
+    .field("y").greaterThan(4)
+    .field("z").greaterThan(10);
+```
+
+This generates an implicit `and` across the field comparisons.
+
+## Text Searching
+
+Morphia also supports MongoDB's text search capabilities.  In order to execute a text search against a collection, the collection must 
+have a [text index]({{< docsref "/core/index-text/" >}}) defined first.  Using Morphia that definition would look like this:
+
+```java
+@Indexes(@Index(fields = @Field(value = "$**", type = IndexType.TEXT)))
+public static class Greeting {
+    @Id
+    private ObjectId id;
+    private String value;
+    private String language;
+    
+    ...
+}
+```
+
+The `$**` value tells MongoDB to create a text index on all the text fields in a document.  A more targeted index can be created, if 
+desired, by explicitly listing which fields to index.  Once the index is defined, we can start querying against it like this [test]({{< 
+srcref "morphia/src/test/java/org/mongodb/morphia/query/TestTextSearching.java">}}) does:
+
+```java
+morphia.map(Greeting.class);
+datastore.ensureIndexes();
+
+datastore.save(new Greeting("good morning", "english"));
+datastore.save(new Greeting("good afternoon", "english"));
+datastore.save(new Greeting("good night", "english"));
+datastore.save(new Greeting("good riddance", "english"));
+
+datastore.save(new Greeting("guten Morgen", "german"));
+datastore.save(new Greeting("guten Tag", "german"));
+datastore.save(new Greeting("guten Nacht", "german"));
+
+List<Greeting> good = datastore.createQuery(Greeting.class)
+                             .search("good")
+                             .order("_id")
+                             .asList();
+Assert.assertEquals(4, good.size());
```

Comment:
I use it also in the snippet below.  I can remove them but I'm trying to at least show a comparison against the expected values...